x-pack/plugins/observability_solution/infra/server/routes/infra/lib/host/get_all_hosts.ts

```diff
@@ -96,15 +85,55 @@ const createQuery = (params: GetInfraMetricsRequestBodyPayload, hostNamesShortLi
                 ],
                 size: 1,
                 sort: {
-                  '@timestamp': 'desc' as const,
+                  '@timestamp': 'desc',
                 },
               },
             },
           },
         },
       },
     },
-  };
+  });
+
+  const monitoredHosts = new Set(
+    response.aggregations?.monitoredHosts.names.buckets.map((p) => p.key) ?? []
+  );
+
+  const result = (response.aggregations?.allHostsMetrics.buckets ?? [])
```

Comment:
Adding some tests would be nice to have here

Reply:
We have functional tests for this endpoint: https://github.com/crespocarlos/kibana/blob/188756-hosts-view-apm-system-hosts-only/x-pack/test/api_integration/apis/metrics_ui/infra.t#L104. I think it covers most of the stuff except for APM hosts.